src/App.js

```diff
@@ -1,11 +1,53 @@
-import React from "react"
+import React, { useEffect } from "react"
+import { Route, Switch } from "react-router-dom"
+import { connect } from "react-redux"
+
+import Home from "./pages/Home"
+import SignIn from "./pages/SignIn"
+import SignUp from "./pages/SignUp"
+import Settings from "./pages/Settings"
+import Editor from "./pages/Editor"
+import Article from "./pages/Article"
+import Profile from "./pages/Profile"
+import ProfileFavorites from "./pages/ProfileFavorites"
+
+import Header from "./components/Header"
+import Footer from "./components/Footer"
+
+import userAPI from "./lib/api/user"
+
+const mapState = (state) => state
+
+function App({ user }) {
+  useEffect(() => {
+    if (user.token) {
+      async function Auth() {
+        try {
+          await userAPI.currentUser(user.token)
+        } catch (error) {
+          console.error(error)
+        }
```

Comment:
Mozesz przeniesc to do `userSlice. Do tego mozesz w state tego slice dac `isLoading` i `error` i na podstawie tego wyswietlac loader jak request idzie do serwera i jakis komunikat jak nie uda sie zalogowac (zamiast console.error). Wiecej info w `src/components/Home/ArticlesList.jsx`